tests/bot/exts/moderation/test_silence.py

```diff
@@ -405,13 +610,22 @@ async def test_sent_correct_message(self):
             (False, silence.MSG_UNSILENCE_MANUAL, PermissionOverwrite(send_messages=False)),
             (False, silence.MSG_UNSILENCE_MANUAL, PermissionOverwrite(add_reactions=False)),
         )
+
         for was_unsilenced, message, overwrite in test_cases:
             ctx = MockContext()
-            with self.subTest(was_unsilenced=was_unsilenced, message=message, overwrite=overwrite):
+
+            for target in [None, MockTextChannel()]:
```

Comment:
You can reduce nesting by using `zip` instead of a nested loop. Do you think this will look nicer?

Reply:
The challenge with that is we need to use two items from the target list, for each item of the test_cases. I'm struggling to see how I could solve that with zip. One solution would be to just double the test_cases list, but that would mean the test runner is completely dependent on there being an odd number of cases.

Reply:
You're right, `zip` won't work. What's needed is a Cartesian product (`itertools.product`). Up to you if you want to make that change.

Reply:
That does help with the nesting, thanks. 27223522